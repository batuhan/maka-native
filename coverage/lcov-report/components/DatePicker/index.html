<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/DatePicker</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/DatePicker
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.68% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.68% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/19</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="DatePicker.android.tsx"><a href="DatePicker.android.tsx.html">DatePicker.android.tsx</a></td>
	<td data-value="63.64" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 63%;"></div><div class="cover-empty" style="width:37%;"></div></div></td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="11" class="abs medium">7/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="11" class="abs medium">7/11</td>
	</tr>

<tr>
	<td class="file high" data-value="DatePicker.ios.tsx"><a href="DatePicker.ios.tsx.html">DatePicker.ios.tsx</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 16 2019 21:39:50 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
