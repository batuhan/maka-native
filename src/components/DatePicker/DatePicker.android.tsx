import * as React from "react";
import { useState } from "react";

import {
  TouchableOpacity,
  TouchableOpacityProps,
  Text,
  DatePickerAndroid
} from "react-native";

interface IProps {
  TouchableOpacityProps?: TouchableOpacityProps;
  defaultDate?: Date;
  onDateChange: (date: Date) => void;
}

// TODO: refactor here
interface DatePickerProps {
  action: string;
  year: number;
  month: number;
  day: number;
}

export default function DatePicker(props: IProps) {
  const [date, setDate] = useState(new Date());
  const { defaultDate, onDateChange } = props;
  const openPicker = async () => {
    try {
      // TODO: refactor here
      const { action, year, month, day } = await DatePickerAndroid.open({
        date: defaultDate || date
      });
      const selectedDate = new Date(year, month, day);
      setDate(selectedDate);
      if (action !== DatePickerAndroid.dismissedAction) {
        onDateChange(selectedDate);
      }
    } catch ({ code, message }) {
      console.warn("Cannot open date picker", message);
    }
  };
  return (
    <TouchableOpacity onPress={openPicker}>
      <Text>{date.toDateString()}</Text>
    </TouchableOpacity>
  );
}
