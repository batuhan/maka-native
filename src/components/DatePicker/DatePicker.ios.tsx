import * as React from "react";
import { useState } from "react";

import {
  TouchableOpacityProps,
  Text,
  TouchableOpacity,
  DatePickerIOSProps,
  DatePickerIOS
} from "react-native";

interface IProps {
  date: Date;
  TouchableOpacityProps?: TouchableOpacityProps;
  defaultDate?: Date;
  DatePickerIOSProps: DatePickerIOSProps;
  onDateChange: (date: Date) => void;
}

export default function DatePicker(props: IProps) {
  const [date, setDate] = useState(props.date || new Date());
  const [isOpenPicker, togglePicker] = useState(false);

  const { onDateChange } = props;
  const onSelectDate = (date: Date) => {
    onDateChange(date);
  };
  const openPicker = () => {
    togglePicker(true);
  };
  return (
    <TouchableOpacity onPress={openPicker}>
      <Text>{date.toDateString()}</Text>
      {isOpenPicker && (
        <DatePickerIOS date={date} onDateChange={onSelectDate} />
      )}
    </TouchableOpacity>
  );
}
