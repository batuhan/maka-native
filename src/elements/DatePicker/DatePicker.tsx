import React from 'react';

import DatePickerAndroid from './DatePicker.android';
import DatePickerIos from './DatePicker.ios';

export default function DatePicker({ defaultDate, onDateChange }: { defaultDate?: Date, onDateChange: (date: Date) => void }) {
  return (
    <>
      <DatePickerAndroid defaultDate={defaultDate} onDateChange={onDateChange} />
      <DatePickerIos defaultDate={defaultDate} onDateChange={onDateChange} />
    </>
  )
}
