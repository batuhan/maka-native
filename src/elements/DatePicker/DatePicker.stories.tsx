import React from 'react'
import { storiesOf } from '@storybook/react-native'

import DatePicker from './DatePicker'

storiesOf('DatePicker', module)
  .add('DatePicker', () => (
    <DatePicker defaultDate={new Date()} onDateChange={value => console.log(value)} />
  ))

