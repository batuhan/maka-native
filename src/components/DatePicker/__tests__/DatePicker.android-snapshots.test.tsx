import React from "react";
import renderer from "react-test-renderer";

import DatePicker from "../DatePicker.android";

describe("DatePickerAndroid", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const onDateChange = jest.fn();
      const tree = renderer.create(<DatePicker onDateChange={onDateChange} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
