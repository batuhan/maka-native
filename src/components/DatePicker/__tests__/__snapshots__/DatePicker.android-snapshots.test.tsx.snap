// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerAndroid Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Thu May 16 2019
  </Text>
</View>
`;
